esmvalcore/cmor/fix.py

```diff
@@ -161,27 +160,13 @@ def fix_metadata(
         session=session,
         frequency=frequency,
     )
-    fixed_cubes = CubeList()
 
-    # Group cubes by input file and apply all fixes to each group element
-    # (i.e., each file) individually
-    by_file = defaultdict(list)
-    for cube in cubes:
-        by_file[cube.attributes.get("source_file", "")].append(cube)
+    for fix in fixes:
```

Comment:
```suggestion
    cubes = CubeList(cubes)
    for fix in fixes:
```

To be backwards-compatible. This currently breaks some fix functions (e.g., [here](https://github.com/ESMValGroup/ESMValCore/blob/1cf3862488e567573d7f9818937cfcf49b01a9fb/esmvalcore/cmor/_fixes/cmip5/gfdl_esm2g.py#L10))